### ISSUE TITLE
Client crashes when broker sends an empty message

### ISSUE THREAD

**Issue by dns2utf8-novaziun** [wez/mosquitto-rs#7]
Hi all

I know it is uncommon, but if the broker sends an empty payload message the client crashes in the rust part.
Should it just ignore the message or deliver an empty?

Cheers,
Stefan

**Comment by wez** [comment]
It shouldn't crash. Can you share the stack trace for this?

**Comment by dns2utf8-novaziun** [comment]
Certainly, is this detailed enough?

```rust
2024-05-08T13:40:53.107479Z DEBUG mosquitto_rs::lowlevel: Client myclient received PUBLISH (d0, q0, r0, m0, 'mytopic/myclient/status', ... (0 bytes))
thread '<unnamed>' panicked at library/core/src/panicking.rs:156:5:
unsafe precondition(s) violated: slice::from_raw_parts requires the pointer to be aligned and non-null, and the total size of the slice not to exceed `isize::MAX`
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread caused non-unwinding panic. aborting.
Aborted
```

Thank you!

**Comment by wez** [comment]
Can you run with RUST_BACKTRACE=1 set in the environment? That will show where the issue comes from.
Note that that precondition check is new in rust 1.78, so a workaround is to use rust 1.77

**Comment by dns2utf8-novaziun** [comment]
99% of the time I expect there to be a message in the payload, but I would love to fix that edge case anyways :)

```rust
2024-05-08T13:48:10.965174Z DEBUG mosquitto_rs::lowlevel: Client myclient received PUBLISH (d0, q0, r0, m0, 'mytopic/myclient/status', ... (0 bytes))
thread '<unnamed>' panicked at library/core/src/panicking.rs:156:5:
unsafe precondition(s) violated: slice::from_raw_parts requires the pointer to be aligned and non-null, and the total size of the slice not to exceed `isize::MAX`
stack backtrace:
   0: rust_begin_unwind
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:645:5
   1: core::panicking::panic_nounwind_fmt::runtime
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/panicking.rs:110:18
   2: core::panicking::panic_nounwind_fmt
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/panicking.rs:123:9
   3: core::panicking::panic_nounwind
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/panicking.rs:156:5
   4: core::slice::raw::from_raw_parts::precondition_check
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/intrinsics.rs:2799:21
   5: core::slice::raw::from_raw_parts
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/slice/raw.rs:98:9
   6: mosquitto_rs::lowlevel::CallbackWrapper<T>::message::{{closure}}
             at /home/stefan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/mosquitto-rs-0.11.1/src/lowlevel.rs:703:17
   7: mosquitto_rs::lowlevel::with_transient_client
             at /home/stefan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/mosquitto-rs-0.11.1/src/lowlevel.rs:633:5
   8: mosquitto_rs::lowlevel::CallbackWrapper<T>::message
             at /home/stefan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/mosquitto-rs-0.11.1/src/lowlevel.rs:695:9
   9: handle__publish
             at /home/stefan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libmosquitto-sys-0.2.3/mosquitto/lib/handle_publish.c:124:5
  10: handle__packet
             at /home/stefan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libmosquitto-sys-0.2.3/mosquitto/lib/read_handle.c:51:11
  11: packet__read
             at /home/stefan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libmosquitto-sys-0.2.3/mosquitto/lib/packet_mosq.c:516:7
  12: mosquitto_loop_read
             at /home/stefan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libmosquitto-sys-0.2.3/mosquitto/lib/loop.c:400:9
  13: mosquitto_loop
             at /home/stefan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libmosquitto-sys-0.2.3/mosquitto/lib/loop.c:151:10
  14: mosquitto_loop_forever
             at /home/stefan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libmosquitto-sys-0.2.3/mosquitto/lib/loop.c:269:9
  15: mosquitto__thread_main
             at /home/stefan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libmosquitto-sys-0.2.3/mosquitto/lib/thread_mosq.c:130:3
  16: <unknown>
  17: <unknown>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread caused non-unwinding panic. aborting.
fish: Job 1, 'RUST_BACKTRACE=1 cargo run' terminated by signal SIGABRT (Abort)
```

**Comment by wez** [comment]
Please try pointing to the git repo to try `main`; if you confirm that that resolves it for you, I'll bump the version and publish to crates.io

**Comment by wez** [comment]
Thanks; published 0.11.2!
